src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -390,7 +391,7 @@ def update_lr_schedulers(self, interval: str, update_plateau_schedulers: bool) -
         if interval == "step" and self._should_accumulate():
             return
         active_optimizers = _get_active_optimizers(
-            self.trainer.optimizers, self.trainer.optimizer_frequencies, self.total_batch_idx
+            self.trainer.optimizers, self.trainer.optimizer_frequencies, self.batch_idx
```

Comment:
Could you add a test to verify this works properly ?

Reply:
I modified the third case of `test_step_scheduling_for_multiple_optimizers_with_frequency` so that it tests that